### DISCUSSION TITLE
Screenshots and cookies dialog

### DISCUSSION THREAD

**Discussion by ghost** [hardkoded/puppeteer-sharp#2018] [General]
Hello, i need to make screenshots of some websites, but what is the problem is EU cookie confirmation dialogs. On some websites when i will take screenshot, they are sometimes on full screen which is, problem :) 

I was thinking about the possibility to use a custom DNS resolver and maybe use some filtering list. But I have no idea how this could be done on puppeteer-sharp. Can anyone think of a solution how this could be done? Maybe dns layer for this cookie dialogs will not be enough. I don't know. 
Or before load that clean it up through this [letter ](https://github.com/tickmao/ADrules/blob/master/I-Dont-Care-About-Cookies.txt) somehow. But I have no idea how it could be done. 
Thank you if you will have some idea

**Comment by kblok** [comment]
Most people try to dismiss the popup as if they were normal users.

**Reply by ghost** [reply]
Yes, but this is not possible if I am browsing webpages generated by the crawler and I want to take a screenshot of them via puppeteer.  There is no user interaction :)  https://www.puppeteersharp.com/examples/Page.ScreenshotAsync.html

**Reply by pnr32** [reply]
@msavara, You can use javascript that will detect and hide cookie confirmation dialogs according to the list of rules. 
For example:
- CSS z-index property with maximum value;
- matching substrings: "cookies", "privacy rules" in innerText;
- matching substring "cookies" in the id attribute;
- etc...

**Reply by ghost** [reply]
I'm not sure this will work. The screenshot is taken internally in puppeteersharp. I am not able to include anything in it at all or to see the html to save before the screenshot is taken I feel. So the screenshot process is encapsulated.

**Comment by ghost** [comment]
Thank you for moving this to the discussion. Now no one will even care now :D

**Reply by kblok** [reply]
It is better than an issue closed, right? :)
Honestly, it's not an issue with the library itself. There is nothing that puppeteer can do.
